Mathlib/Analysis/Polynomial/Height.lean

```diff
@@ -0,0 +1,103 @@
+/-
+Copyright (c) 2025 Kevin H. Wilson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kevin H. Wilson
+-/
+module
+
+public import Mathlib.Analysis.Complex.Polynomial.Basic
+
+/-!
+# (Naive) height of polynomials
+
+In this file we define the (naive) height of a polynomial and prove some basic properties.
+See its relationships with Mahler measure in `Mathlib/Analysis/Polynomial/MahlerMeasure.lean`.
+
+## Main definitions
+
+- `Polynomial.height p`: the (naive) height of a polynomial, equal to the maximum norm of its
+  coefficients (or zero for the zero polynomial)
+-/
+
+@[expose] public section basic
+
+variable {A : Type*} [SeminormedRing A] (p : Polynomial A) {a : A}
+
+namespace Polynomial
+
+/-- The (naive) height of a polynomial, defined as the maximum of its coefficients (or `0` when
+`p = 0`) -/
+noncomputable def height : ℝ :=
```

Comment:
It feels weird to special-case the zero polynomial. Why not just write `⨆ i, ‖p.coeff i‖`, which should work in all cases?